pywt/tests/test_modes.py

```diff
@@ -61,6 +65,31 @@ def test_dwt_idwt_allmodes():
         assert_allclose(pywt.idwt(cA, cD, 'db2', mode), x, rtol=1e-10)
 
 
+def test_dwt_short_input_allmodes():
+    # some test cases where the input is shorter than the DWT filter
+    x = [1, 2, 3]
+    wavelet = 'db2'
+    # manually pad each end by the filter size (4 for 'db2' used here)
+    padded_x = {'zero': [0, 0, 0, 0, 1, 2, 3, 0, 0, 0, 0],
+                'constant': [1, 1, 1, 1, 1, 2, 3, 3, 3, 3, 3],
+                'symmetric': [3, 3, 2, 1, 1, 2, 3, 3, 2, 1, 1],
+                # 'reflect': [1, 2, 3, 2, 1, 2, 3, 2, 1, 2, 3],
```

Comment:
should not have been commented.  The test was failing in that case and I need to fix it.

Basically, these are tests for when the input signal length is less than the wavelet filter length.  That case (likely pretty rare in practice) is not currently covered by the Matlab tests (they start with signals of size w.dec_len)